### PR TITLE
[MINOR][DOCS] Fix R documentation generation instruction for roxygen2

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

This PR proposes to fix `roxygen2` to `5.0.1` in `docs/README.md` for SparkR documentation generation.

If I use higher version and creates the doc, it shows the diff below. Not a big deal but it bothered me.

```diff
diff --git a/R/pkg/DESCRIPTION b/R/pkg/DESCRIPTION
index 855eb5bf77f..159fca61e06 100644
--- a/R/pkg/DESCRIPTION
+++ b/R/pkg/DESCRIPTION
@@ -57,6 +57,6 @@ Collate:
     'types.R'
     'utils.R'
     'window.R'
-RoxygenNote: 5.0.1
+RoxygenNote: 6.0.1
 VignetteBuilder: knitr
 NeedsCompilation: no
```

## How was this patch tested?

Manually tested. I met this every time I set the new environment for Spark dev but I have kept forgetting to fix it.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @felixcheung could you take a look when you are available?

**Review by felixcheung — APPROVED** [review]
thats fine but I thought Jenkins is running 6.0.1? or maybe only on some boxes

File: docs/README.md [hunk 1]
```diff
@@ -22,10 +22,13 @@ $ sudo gem install jekyll jekyll-redirect-from pygments.rb
...
 $ sudo pip install sphinx pypandoc mkdocs
-$ sudo Rscript -e 'install.packages(c("knitr", "devtools", "roxygen2", "testthat", "rmarkdown"), repos="http://cran.stat.ucla.edu/")'
+$ sudo Rscript -e 'install.packages(c("knitr", "devtools", "testthat", "rmarkdown"), repos="http://cran.stat.ucla.edu/")'
+$ sudo Rscript -e 'devtools::install_version("roxygen2", version = "5.0.1", repos="http://cran.stat.ucla.edu/")'
 ```
 
-(Note: If you are on a system with both Ruby 1.9 and Ruby 2.0 you may need to replace gem with gem2.0)
+Note: If you are on a system with both Ruby 1.9 and Ruby 2.0 you may need to replace gem with gem2.0.
+
+Note: Other versions of roxygen2 might work in SparkR documentation generation but `RoxygenNote` field in `SPARK_HOME/R/pkg/DESCRIPTION` is 5.0.1, which is updated if the version is mismatched.
```
Comment (felixcheung) [inline docs/README.md]: nit `$SPARK_HOME/R/pkg/DESCRIPTION`?

Comment (HyukjinKwon) [reply docs/README.md]: Yea ... but this format was used below too - https://github.com/HyukjinKwon/spark/blob/a0bba18ef9a19774e1965867241f56fb0508c579/docs/README.md#api-docs-scaladoc-javadoc-sphinx-roxygen2-mkdocs

Let me change it to `$SPARK_HOME` all. I believe that's more correct.

**Comment by HyukjinKwon** [comment]
> thats fine but I thought Jenkins is running 6.0.1? or maybe only on some boxes

I am not sure of the specific version but I assume it's older version ..

**Review by felixcheung — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master and branch-2.3.

Thanks @felixcheung for reviewing this.

**Pull request closed without merging**